lib/path.js

```diff
@@ -77,7 +77,7 @@ function normalizeString(path, allowAboveRoot, separator, isPathSeparator) {
               lastSegmentLength = 0;
             } else {
               res = res.slice(0, lastSlashIndex);
-              lastSegmentLength = res.length - 1 - res.lastIndexOf(separator);
+              lastSegmentLength = res.length - 1 - lastSlashIndex;
```

Comment:
This is not identical to the former. `res` is reassigned in the line above and it does not contain `lastSlashIndex` anymore.

Reply:
Oops, you are right, thanks.

Totally a foolish mistake.